File: meteorological-assessment-tracker-ui/src/app/tracker/models/area-data.model.ts
import { DayLightRecordModel } from './day-light.model';
import { TideRecordModel } from './tide.model';
import { RainfallRecordModel } from './Rainfall.model';
import { WeatherRecordModel } from './weather.model';


export class AreaDataModel {
    Date: Date;
    AreaName: string;
    Daylight: DayLightRecordModel;
    Tide: TideRecordModel[];
    Weather: WeatherRecordModel[];
    Rainfall: RainfallRecordModel[];
}

File: meteorological-assessment-tracker-data/DataTransformerApi/Models/AreaDataModel.cs
using System;
using System.Collections.Generic;

namespace DataTransformerApi.Models
{
    public class AreaDataModel
    {
        public DateTime Date { get; set; }
        public DaylightModel Daylight { get; set; }
        public List<RainfallModel> Rainfall { get; set; }
        public List<TideModel> Tide { get; set; }
        public List<WeatherModel> Weather { get; set; }
    }
}